From 70a4a971f4b46e80d288abe413adb6e2876d8f71 Mon Sep 17 00:00:00 2001
From: Danny Al-Gaaf <danny.al-gaaf@bisect.de>
Date: Tue, 14 May 2013 16:50:57 +0200
Subject: [PATCH] mds/Server.cc: fix possible NULL pointer dereference

CID 1019555 (#1 of 1): Dereference after null check (FORWARD_NULL)
  var_deref_model: Passing null pointer "in" to function
  "Server::_need_force_journal(CInode *, bool)", which dereferences it.

Signed-off-by: Danny Al-Gaaf <danny.al-gaaf@bisect.de>
---
 src/mds/Server.cc |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/mds/Server.cc b/src/mds/Server.cc
index d97047d..c9fac42 100644
--- a/src/mds/Server.cc
+++ b/src/mds/Server.cc
@@ -6943,7 +6943,7 @@ void Server::do_rename_rollback(bufferlist &rbl, int master, MDRequest *mdr)
   bool force_journal_dest = false;
   if (in && in->is_dir() && srcdn->authority().first != whoami)
     force_journal_src = _need_force_journal(in, false);
-  if (target && target->is_dir())
+  if (in && target && target->is_dir())
     force_journal_dest = _need_force_journal(in, true);
   
   version_t srcdnpv = 0;
-- 
1.7.2.5

